src/library/browse/browsetablemodel.cpp

```diff
@@ -210,8 +210,8 @@ void BrowseTableModel::addSearchColumn(int index) {
 }
 
 void BrowseTableModel::setPath(mixxx::FileAccess path) {
-    if (path.info().hasLocation()) {
-        m_currentDirectory = path.info().locationPath();
+    if (path.info().hasLocation() && path.info().isDir()) {
+        m_currentDirectory = path.info().location();
```

Comment:
It turns out that location() does not strip a trailing slash. Is this new behaviour desired? 
I have hacked up a test to verify this: 
https://github.com/daschuer/mixxx/blob/2114f3e34122b195bde868e2a3355f9626324bf6/src/test/fileinfo_test.cpp#L62

Reply:
Yes, intended:
> Since `locationPath()` was chosen only to remove the trailing / from the paths passed by the BrowseFeature to have a 'clean' model key, and working around the sideeffect caused by the Rec path (no trailing /)...
I changed setPath() to use the entire path location (with /) for the model key and avoid the Rec issue in the first place: